cg/apps/lims/api.py

```diff
@@ -85,74 +85,114 @@ def _export_artifact(self, lims_artifact):
 
     def get_received_date(self, lims_id: str) -> str:
         """Get the date when a sample was received."""
-        lims_artifacts = self.get_artifacts(process_type='CG002 - Reception Control',
-                                            samplelimsid=lims_id)
-        for artifact in lims_artifacts:
-            udf_key = 'date arrived at clinical genomics'
-            if artifact.parent_process and artifact.parent_process.udf.get(udf_key):
-                return artifact.parent_process.udf.get(udf_key)
+
+        step_names_udfs = MASTER_STEPS_UDFS['received_step']
+        received_dates = []
+
+        for process_type in step_names_udfs:
+            artifacts = self.get_artifacts(process_type=process_type, samplelimsid=lims_id)
+
+            for artifact in artifacts:
+                udf_key = step_names_udfs[process_type]
+                if artifact.parent_process and artifact.parent_process.udf.get(udf_key):
+                    received_dates.append((artifact.parent_process.date_run,
+                                           artifact.parent_process.udf.get(udf_key)))
+
+        sorted_dates = sorted(received_dates, key=lambda x: x[0])
```

Comment:
I have a hard time understanding this line, can we "hide" the sorting in a method call? something like get_dates_sorted_ascending(received_dates) or get_dates_sorted_descending(received_dates) so I can understand what is going on without having to understand how the sort function works

Reply:
created method `sort_by_date_run` in commits https://github.com/Clinical-Genomics/cg/pull/376/commits/941460dad627b65af89ea27e0174ddeebb095c5a and https://github.com/Clinical-Genomics/cg/pull/376/commits/f1b6b03e325ed8767671385beadc493d1f6bbb6b, made it really general so it will sort both dates and methods based on `date_run`. Thoughts?

Reply:
Great work of art man! I suggest we call sort_by_date() from within most_recent_date() since most_recent_date currently has a requirement on the list it is passed, yes? It needs the list to be sorted in a specific order to work, right? If we sort the list within the method then that requirement disapper. What do you think about it?

Reply:
Good idea, fixed in https://github.com/Clinical-Genomics/cg/pull/376/commits/d1c5b7f33a1b83180e16e8d67eb0ddd95f56558f